src/app/modules/account/pages/spouse-info/add-spouse/add-spouse.component.ts

```diff
@@ -58,7 +58,9 @@ export class AddSpouseComponent extends Base implements OnInit {
   }
 
   get marriageDateError() {
-    return this.spouse.dateOfBirth > this.spouse.marriageDate ? true :false
+    return this.spouse.dateOfBirth !== undefined && this.spouse.marriageDate !== undefined
+      && this.spouse.dateOfBirth !== null && this.spouse.marriageDate !== null
+      && this.spouse.dateOfBirth > this.spouse.marriageDate;
```

Comment:
Are you sure that `date > otherDate` have the desired effect? I've always used the `isAfter(...)`/`isBefore(...)` helpers from the `date-fns` library.

Reply:
I've updated the date check to use isAfter()